dbms/tests/queries/0_stateless/00950_column_encoding_double_delta.sql

```diff
@@ -0,0 +1,211 @@
+DROP TABLE IF EXISTS reference;
+DROP TABLE IF EXISTS doubleDelta;
+
+CREATE TABLE reference (
+    key      UInt64,
+    valueU64 UInt64,
+    valueU32 UInt32,
+    valueU16 UInt16,
+    valueU8  UInt8,
+    valueI64 Int64,
+    valueI32 Int32,
+    valueI16 Int16,
+    valueI8  Int8,
+    valueDT  DateTime,
+    valueD   Date
+) Engine = MergeTree ORDER BY key;
+
+
+CREATE TABLE doubleDelta (
+    key      UInt64   CODEC(DoubleDelta),
+    valueU64 UInt64   CODEC(DoubleDelta),
+    valueU32 UInt32   CODEC(DoubleDelta),
+    valueU16 UInt16   CODEC(DoubleDelta),
+    valueU8  UInt8    CODEC(DoubleDelta),
+    valueI64 Int64    CODEC(DoubleDelta),
+    valueI32 Int32    CODEC(DoubleDelta),
+    valueI16 Int16    CODEC(DoubleDelta),
+    valueI8  Int8     CODEC(DoubleDelta),
+    valueDT  DateTime CODEC(DoubleDelta),
+    valueD   Date     CODEC(DoubleDelta)
+) Engine = MergeTree ORDER BY key;
+
+
+-- n^3 covers all double delta storage cases, from small difference between neighbour values (stride) to big.
+INSERT INTO reference (key, valueU64, valueU32, valueU16, valueU8, valueI64, valueI32, valueI16, valueI8, valueDT, valueD)
+	SELECT number as n, n * n * n as v, v, v, v, v, v, v, v, toDateTime(v), toDate(v) FROM system.numbers LIMIT 1, 100;
+
+-- best case - constant stride
+INSERT INTO reference (key, valueU64, valueU32, valueU16, valueU8, valueI64, valueI32, valueI16, valueI8, valueDT, valueD)
+	SELECT number as n, n as v, v, v, v, v, v, v, v, toDateTime(v), toDate(v) FROM system.numbers LIMIT 101, 100;
+
+-- checking for overflow
```

Comment:
Overflow test cases

Reply:
18446744073709551616 is String to UInt64 overflow. I expected to see 0xffffffffffffffff, 0x7fffffffffffffff,0x8000000000000000 in cases where overflow is inside compression func not in input phase. Ex. 
```
0x7fffffffffffffff - (-1)
0x7fffffffffffffff - 0x8000000000000000
```

Reply:
This issue was addresses in unit tests.